src/Client.php

```diff
@@ -951,19 +981,35 @@ public function send(Request $request = null)
             }
 
             if ($this->config['outputstream']) {
+                if (! $adapter instanceof StreamInterface) {
+                    throw new Client\Exception\RuntimeException('Adapter does not support streaming');
+                }
+
                 if ($stream === null) {
                     $stream = $this->getStream();
                     if (! is_resource($stream) && is_string($stream)) {
-                        $stream = fopen($stream, 'r');
+                        $stream = fopen($stream, 'rb');
                     }
                 }
+
+                if (! \is_resource($stream)) {
+                    throw new UnexpectedValueException('Stream is not a resource');
+                }
+
                 $streamMetaData = stream_get_meta_data($stream);
                 if ($streamMetaData['seekable']) {
                     rewind($stream);
                 }
+
                 // cleanup the adapter
                 $adapter->setOutputStream(null);
                 $response = Response\Stream::fromStream($response, $stream);
+
+                // streamName can just be a string at this point
+                if (! \is_string($this->streamName)) {
+                    throw new UnexpectedValueException('Unexpected value for streamName');
```

Comment:
It's not a possible behavior,  so it can't be tested. That's why is an `UnexpectedValueException`.

We can ignore it on phpstan or we can override the value